src/System.Private.CoreLib/shared/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Float.cs

```diff
@@ -61,29 +61,6 @@ public static bool TryFormat(float value, Span<byte> destination, out int bytesW
 
         private static bool TryFormatFloatingPoint<T>(T value, Span<byte> destination, out int bytesWritten, StandardFormat format) where T : IFormattable, ISpanFormattable
         {
-            if (format.IsDefault)
-            {
-                format = 'G';
```

Comment:
Will the behavior remain as expected when `format` is default?

It seems like if `format == 0` and `precision == 0`, the `formatText` is just the empty span. If the default behavior is correct, should we still special case `format.IsDefault` to avoid the stackalloc/call to `format.Format`?

Reply:
We could probably special-case it to just directly do `formatText = string.Empty`, but the current behavior results in basically the same.

Reply:
Modified to avoid the `format.Format` call...

Can't avoid the stackalloc due to span safety rules.

Reply:
> Can't avoid the stackalloc due to span safety rules.

Why not?  You can't make it `Span<char> formatText = stackalloc char[0];`?

Reply:
Looks like doing `stackalloc char[0]` works, I've updated the PR.

Seems like a strange limitation that it works for this scenario (which is example `D`), but not for `A`, `B`, or `C`:
https://sharplab.io/#v2:EYLgHgbALAPgAgBgARwIwG4CwAoOBmFAJiQFkBPAYQBsBDAZzqQG8ck2UC4okBBACmAB7QVSQBjQQDsAJgEsALrKkBKZq3YaAygAcakgDxiAFjQBOAPiQAzQaYC2NeVmwaN610llWkfCTIVKksruriwuHq429o5IALxI0gCmVjQArlROIRoAvlnsiVR0iex5bGERkbYO8nFIdPI0YgDWNFRUgmLiJqYA2ngAus4VGlHVtaOOAHSaVLJiiXwIADRIhMpDEbnhbFtu2xwo3ABCAsKifnKKKmr7WroGxmaWE5m3JW9sXj4XASqlN8N2C9avVGi02h0umYeghBv9dhECkV3hVyoDrFUYvFQc1Wu1Oo9egMNoDgfEXtNZvNFis1iTXAidiEQvhDkgKKcROIpJdAqo0R4dHpDN1npiavEkil0q8Kv8vnwAIQ/K5Bf4C4ZkuoNXEQgndPpwj6VaISjGmylzBbLVbreEhRlIFmcbgAEU55x5vyCAIiQoeovNY2xOvB+KhvVh9L2FQVyq9quCxo1FS1OLDkMJhujES1FJmVpptpzTP2W2yQA=

Reply:
I guess the compiler can't reason about stackalloc with full human fidelity :)

`B` is interesting since all branches indicate the span is stackalloc'd. I can see why `A` and `C` are limiting since in some conditions the span can be returned (when not stackalloc'd), in others, it can't.

cc @jaredpar, @agocke

Reply:
> I can see why `A` and `C` are limiting

The compiler actually optimizes `stackalloc char[0]` to `default`, so I'm not so sure (and I would think would be able to just always treat the local as the "most restricted" kind for analysis purposes) :)

Reply:
Does Roslyn need an issue opened?